### PR TITLE
preparations for search feature

### PR DESCRIPTION
**Pull request by cre4ture**
refactoring only: extract ChunkID from ChunkCache header
I (slowly) started with bringing the search functionality to master (#199). But this needs some preparation steps. As this work will go over a timeframe of potentially months I want to push any working version in between to master as soon as possible to avoid merging conflicts later. Hope you are not annoyed by many smaller pull requests that doesn't bring functionality nor fixes.

### CODE REVIEW

File: chunkid.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#ifndef CHUNKID_H
+#define CHUNKID_H
+
+// ChunkID is the key used to identify entries in the Cache
+// Chunks are identified by their coordinates (CX,CZ) but a single key is needed to access a map like structure
+class ChunkID {
+ public:
+  ChunkID(int cx, int cz);
+  bool operator==(const ChunkID &) const;
+  friend unsigned int qHash(const ChunkID &);
```
Comment (madmaxoft) [inline chunkid.h]: The previous declaration returned an `uint`; with bitwise operators that's the only safe way, some operations are undefined behavior on certain signed types. Are you still sure you need an `int`?

Comment (cre4ture) [reply chunkid.h]: I replaced `uint` with `unsigned int` which should be the very same at the end.
take a look into `qglobal.h` and you find this definition of uint: `typedef unsigned int uint;`
I just found it more elegant to not have any dependency to qt here in the header anymore.
But it's rather a subtle difference considering that the most things in minutor make use of Qt stuff ;-). So if you see an other reason why you would prefer `uint`over `unsigned int` I'll happy to change it back - no problem.

Comment (madmaxoft) [reply chunkid.h]: Right, I'm blind, completely missed the `unsigned` there. Sorry.

**Comment by cre4ture** [comment]
I just did an other step and added it. Apperently this also updates the Pull Request automatically when pushed.

**Pull request merged**